scan-command/src/main/java/io/ballerina/scan/internal/ScanCmd.java

```diff
@@ -244,6 +314,19 @@ private Optional<Project> getProject() {
         }
     }
 
+    private URLClassLoader loadPlatformPlugins(List<String> jarPaths) {
+        List<URL> jarUrls = new ArrayList<>();
```

Comment:
Preallocate this list with the expected size
```suggestion
        List<URL> jarUrls = new ArrayList<>(jarPaths.size());
```